DurableFunctionsAnalyzer/DurableFunctionsAnalyzer/DurableFunctionsAnalyzer/SyntaxNodeUtils.cs

```diff
@@ -0,0 +1,178 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using System;
+using System.Linq;
+
+namespace DurableFunctionsAnalyzer
+{
+    public static class SyntaxNodeUtils
+    {
+        public static DurableVersion? version;
+
+        public static DurableVersion GetDurableVersion(SemanticModel semanticModel)
+        {
+            if (version != null)
+            {
+                return (DurableVersion)version;
+            }
+
+            var versionTwoInterface = semanticModel.Compilation.GetTypeByMetadataName("Microsoft.Azure.WebJobs.IDurableOrchestrationContext");
+            version = versionTwoInterface != null ? DurableVersion.V2 : DurableVersion.V1;
+            return (DurableVersion)version;
+        }
+        
+        public static bool IsInsideOrchestrator(SyntaxNode node)
+        {
+            if (TryGetMethodDeclaration(out SyntaxNode methodDeclaration, node))
+            {
+                var parameterList = methodDeclaration.ChildNodes().Where(x => x.IsKind(SyntaxKind.ParameterList)).First();
+
+                foreach (SyntaxNode parameter in parameterList.ChildNodes())
+                {
+                    var attributeList = parameter.ChildNodes().Where(x => x.IsKind(SyntaxKind.AttributeList));
+                    if (attributeList.Count() >= 1 && attributeList.First().ChildNodes().First().ToString().Equals("OrchestrationTrigger"))
+                    {
+                        return true;
+                    }
+                }
+            }
+            return false;
+        }
+
+        internal static bool TryGetClassSymbol(out INamedTypeSymbol classSymbol, SemanticModel semanticModel)
+        {
+            var classDeclarationSyntax = semanticModel.SyntaxTree.GetRoot().DescendantNodes().OfType<ClassDeclarationSyntax>().Last();
+            if (classDeclarationSyntax != null)
+            {
+                var classDeclarationSymbol = semanticModel.GetDeclaredSymbol(classDeclarationSyntax);
+                if (classDeclarationSymbol != null)
+                {
+                    classSymbol = classDeclarationSymbol;
+                    return true;
+                }
+            }
+
+            classSymbol = null;
+            return false;
+        }
+
+        internal static bool TryGetFunctionName(out string functionName, SyntaxNode functionAttribute)
```

Comment:
This is a note for all of your extension methods. In general, the out parameter of `TryGet` methods go last, and the collection/object you are retrieving the item from goes first.